rice-namenode/Leases.cc

```diff
@@ -0,0 +1,103 @@
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <easylogging++.h>
+#include <rpcserver.h>
+#include <ConfigReader.h>
+
+#include "Leases.h"
+
+/**
+ * The implementation of the rpc calls. 
+ */
+namespace lease {
+
+const int Lease::LEASE_EXPIRATION_TIME = 3600; // close file if its been an hour 
+
+Lease::Lease(std::string in) : filename(in) {}
+
+/**
+ * This happens upon a renewal of a lease
+ */
+void Lease::resetTimer() {
+	time = 0;
+}
+
+/**
+ * This happens every LEASE_CHECK_TIME seconds. If the file is
+ * expired, then we forcibly close it. Otherwise, we hadd 60
+ * seconds onto the timer 
+ */
+bool Lease::isExpired(int time) {
+	// TODO close the file and recover the blocks
+	if (time >= LEASE_EXPIRATION_TIME) {
+		return false;
+	}
+	time += time;
```

Comment:
the scoping always gets me turned around. can we rename the local variable so it doesn't look like the class variable?